main/src/library/Core/Io/File.flix

```diff
@@ -31,6 +31,51 @@ namespace Core/Io/File {
         import new java.io.File(String) as newFile;
         File(newFile(s))
 
+    ///
+    /// Combines every string in `xs` into a single path
+    ///
+    pub def combine(n: Nel[String]): File & Impure = {
```

Comment:
I don't understand why you would want something like this, especially with the nel requirement?

Afterall, just a little bit further down you do construct a File with the empty path.

Reply:
It mirrors the [`Paths.get`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Paths.html#get(java.lang.String,java.lang.String...)) static method. It's an alternative to having to repeatedly apply the `File.add` function.

I also think, in the case of the function taking a `List[String]`, it'd be an unusual behavior to say that `combine(Nil) == new(".")` i.e. the current working directory. Specifically because the empty list implies "no path". The current working directory is also not necessarily the base path of the combined so using it as the base in the fold would be non-standard behavior.

As an example the following returns `Ok(File(D:\foo\bar))` on my system
```scala
canonical(combine(Nel("/foo", "bar" :: Nil)))
```

Reply:
OK, I buy you argument. I would still like to rename it. It should be called something that starts with `new` and placed after the ordinary new constructor. Perhaps `newFromNel` or `newFromFragments` or some other thing.

Reply:
I like the sound of `newFromFragments`